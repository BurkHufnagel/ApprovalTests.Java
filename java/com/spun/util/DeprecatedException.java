package com.spun.util;

import com.spun.util.introspection.Caller;

public class DeprecatedException extends FormattedException
{
  public DeprecatedException(String useNewMethod, Object... formattingParams)
  {
    super("%s is Depercated.\n Instead, please use :\n %s", methodName(Caller.get(1)),
        String.format(useNewMethod, formattingParams));
  }
  private static Object methodName(StackTraceElement trace)
  {
    if (trace.getMethodName().equals("<init>")) { return trace.getClassName(); }
    return trace.getMethodName();
  }
}