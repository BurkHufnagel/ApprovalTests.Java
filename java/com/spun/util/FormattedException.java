package com.spun.util;

import com.spun.util.logger.SimpleLogger;

public class FormattedException extends RuntimeException
{
  public FormattedException(String string, Object... params)
  {
    super(String.format(string, params));
    SimpleLogger.variable(this.getMessage());
  }
}
