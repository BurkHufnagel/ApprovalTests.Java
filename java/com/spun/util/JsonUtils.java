package com.spun.util;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonElement;
import com.google.gson.JsonParser;

public class JsonUtils
{
  public static String prettyPrint(String json)
  {
    if (!ObjectUtils.isClassPresent("com.google.gson.Gson")) { throw new RuntimeException(
        "Missing Gson dependency\n  Pretty print uses Gson parser.\n  You can get this from the maven repo \n  or https://github.com/google/gson"); }
    Gson gson = new GsonBuilder().setPrettyPrinting().create();
    JsonParser jp = new JsonParser();
    JsonElement je = jp.parse(json);
    return gson.toJson(je);
  }
}
