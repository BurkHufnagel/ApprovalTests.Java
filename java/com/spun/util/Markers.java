package com.spun.util;

public class Markers implements AutoCloseable
{
  private String text;
  public Markers(String text)
  {
    this.text = text;
    MySystem.markerIn(text);
  }
  @Override
  public void close()
  {
    MySystem.markerOut(text);
  }
}
