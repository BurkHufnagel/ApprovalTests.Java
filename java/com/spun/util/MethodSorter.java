package com.spun.util;

/**
 * @deprecated use  OrderBy.descending(a -> a.length())
 */
@Deprecated
public class MethodSorter<T>
{
  public MethodSorter(Class<? extends T> classType, String methodName, boolean ascending)
  {
    throw new DeprecatedException("OrderBy.%s((%s a) -> a.%s())", ascending ? "ascending" : "descending",
        classType.getName(), methodName);
  }
}
