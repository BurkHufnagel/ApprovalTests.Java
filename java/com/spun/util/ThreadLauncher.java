package com.spun.util;

import java.lang.reflect.Method;

import org.lambda.actions.Action0;

/**
 * @deprecated use  LambdaThreaLauncher( ()-> object.method(params))
 */
public class ThreadLauncher
{
  /***********************************************************************/
  public ThreadLauncher(Object object, Method method, Object[] objectParams, long delay)
  {
    String params = StringUtils.join(objectParams, ",", o -> "" + o);
    throw new DeprecatedException("new LambdaThreadLauncher(()-> object.%s(%s))", method.getName(), params);
  }
  /***********************************************************************/
  public ThreadLauncher(Object object, Method method, Object[] objectParams)
  {
    this(object, method, objectParams, 0);
  }
  /***********************************************************************/
  public ThreadLauncher(Class<?> clazz, String methodName, long delay)
  {
    this(null, ClassUtils.getMethod(clazz, methodName), null, delay);
  }
  /***********************************************************************/
  public ThreadLauncher(Class<?> clazz, String methodName)
  {
    this(null, ClassUtils.getMethod(clazz, methodName), null, 0);
  }
  /***********************************************************************/
  public ThreadLauncher(Object object, String methodName, long delay)
  {
    this(object, ClassUtils.getMethod(object.getClass(), methodName), null, delay);
  }
  /***********************************************************************/
  public ThreadLauncher(Object object, String methodName)
  {
    this(object, ClassUtils.getMethod(object.getClass(), methodName), null, 0);
  }
  /***********************************************************************/
  public ThreadLauncher(Object object, String methodName, Object[] objectParams, long delay)
  {
    this(object, MethodExecutionPath.Parameters.getBestFitMethod(object.getClass(), methodName,
        getClassArray(objectParams)), objectParams, delay);
  }
  /***********************************************************************/
  private static Class[] getClassArray(Object[] objectParams)
  {
    Class[] classes = new Class[objectParams.length];
    for (int i = 0; i < objectParams.length; i++)
    {
      classes[i] = objectParams[i].getClass();
    }
    return classes;
  }
  public static void launch(Action0 action)
  {
    new LambdaThreadLauncher(action);
  }
}
