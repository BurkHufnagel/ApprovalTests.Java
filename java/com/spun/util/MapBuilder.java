package com.spun.util;

import java.util.HashMap;

public class MapBuilder<K, V> extends HashMap<K, V>
{
  public MapBuilder(K key, V value)
  {
    and(key, value);
  }
  public MapBuilder<K, V> and(K key, V value)
  {
    put(key, value);
    return this;
  }
}
