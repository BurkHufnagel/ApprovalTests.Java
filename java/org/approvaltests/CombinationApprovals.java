package org.approvaltests;

import java.util.Arrays;

import org.approvaltests.legacycode.LegacyApprovals;
import org.lambda.functions.Function2;

public class CombinationApprovals
{
  public static void verifyAllCombinations(Object call, String method, Object[]... parametersVariations)
      throws Exception
  {
    LegacyApprovals.LockDown(call, method, parametersVariations);
  }
  public static <IN1, IN2, OUT> void verifyAllCombinations(Function2<IN1, IN2, OUT> call, IN1[] parameters1,
      IN2[] parameters2) throws Exception
  {
    StringBuffer output = new StringBuffer();
    for (IN1 in1 : parameters1)
    {
      for (IN2 in2 : parameters2)
      {
        output.append(String.format("%s => %s \r\n", Arrays.asList(in1, in2), call.call(in1, in2)));
      }
    }
    Approvals.verify(output);
  }
}
