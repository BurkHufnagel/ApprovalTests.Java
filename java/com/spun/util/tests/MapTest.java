package com.spun.util.tests;

import java.util.Date;
import java.util.Map;

import org.approvaltests.Approvals;
import org.junit.Test;

import com.spun.util.MapBuilder;

public class MapTest
{
  @Test
  public void test()
  {
    Map p = params("name", "Llewellyn").and("age", 40).and("date", new Date(2001, 01, 01));
    Approvals.verify(p);
  }
  public static MapBuilder<String, Object> params(String key, Object value)
  {
    return new MapBuilder<String, Object>(key, value);
  }
}
