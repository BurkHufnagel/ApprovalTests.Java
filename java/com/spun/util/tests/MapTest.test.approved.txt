age : 40 
date : Fri Feb 01 00:00:00 EET 3901 
name : Llewellyn 
